lib/matplotlib/axes/_axes.py

```diff
@@ -6281,24 +6281,19 @@ def _normalize_input(inp, ename='input'):
                     xvals.append(x.copy())
                     yvals.append(y.copy())
 
-            if fill:
-                # add patches in reverse order so that when stacking,
-                # items lower in the stack are plottted on top of
-                # items higher in the stack
-                for x, y, c in reversed(list(zip(xvals, yvals, color))):
-                    patches.append(self.fill(
-                        x, y,
-                        closed=True,
-                        facecolor=c,
-                        margins=margins))
-            else:
-                for x, y, c in reversed(list(zip(xvals, yvals, color))):
-                    split = 2 * len(bins)
-                    patches.append(self.fill(
-                        x[:split], y[:split],
-                        closed=False, edgecolor=c,
-                        fill=False,
-                        margins=margins))
+            #stepfill is closed, step is not
+            split = -1 if fill else 2 * len(bins)
+            # add patches in reverse order so that when stacking,
+            # items lower in the stack are plottted on top of
+            # items higher in the stack
+            for x, y, c in reversed(list(zip(xvals, yvals, color))):
+                patches.append(self.fill(
+                    x[:split], y[:split],
+                    closed=True if fill else None,
+                    facecolor=c,
```

Comment:
I suspect this behavior is going to change in 2.x as I think edge color now defaults to face color rather than 'k' so just leave this as-is.